### PR TITLE
Regularization

### PR DESCRIPTION
**Pull request by sladenheim**
This PR revises and formats the regularization lecture.

Some changes:

- introduction section on overfitting
- introduction on regularization in general
- lecture overview
- cleaned up sections on 
   - multicollinearity
   - condition number
   - near linear dependence

### CODE REVIEW

File: ds701_book/jupyter_notebooks/19-Regression-III-More-Linear.ipynb [hunk 1]
```diff
@@ -7,39 +7,27 @@
         "---\n",
```
Comment (trgardos) [inline ds701_book/jupyter_notebooks/19-Regression-III-More-Linear.ipynb:7]: <p>Suggest adding a sentence stating what is the y variable we are trying to predict.</p><p> </p><p>Also, should we define the columns just as an fyi?</p><p> </p><p>TOTEMP - Total Employment</p><p>        GNPDEFL - GNP deflator</p><p>        GNP - GNP</p><p>        UNEMP - Number of unemployed</p><p>        ARMED - Size of armed forces</p><p>        POP - Population</p><p>        YEAR - Year (1947 - 1962)</p><p> </p><p>From</p><p><a href="https://www.statsmodels.org/stable/datasets/generated/longley.html" target="_blank">https://www.statsmodels.org/stable/datasets/generated/longley.html</a></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tools4ds/DS701-Course-Notes/pull/162/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (trgardos) [inline ds701_book/jupyter_notebooks/19-Regression-III-More-Linear.ipynb:7]: <p>Should we comment somewhere that PCA also addresses this issue by transforming the correlated features into uncorrelated features? I suppose the advantage of this regularization approach is that you keep the feature names, so it is more explainable.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tools4ds/DS701-Course-Notes/pull/162/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (trgardos) [inline ds701_book/jupyter_notebooks/19-Regression-III-More-Linear.ipynb:7]: <p>Can we add the 2nd plot that shows the solution like in the previous example? Or would it not fit?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tools4ds/DS701-Course-Notes/pull/162/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (trgardos) [inline ds701_book/jupyter_notebooks/19-Regression-III-More-Linear.ipynb:7]: <p>I think $\hat{\boldsymbol{\beta}} = (X^TX +cI)^{−1}X^T$ is <strong>not</strong> a linear function of X.  </p><p> </p><p>Regarding Scaling -- should we point out that we said in the linear regression lecture that scaling was <strong>not</strong> required. But now for this method, it <strong>is</strong> required again.  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tools4ds/DS701-Course-Notes/pull/162/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (trgardos) [inline ds701_book/jupyter_notebooks/19-Regression-III-More-Linear.ipynb:7]: <p>Can we say a little more about why picked this formula? and what it means?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tools4ds/DS701-Course-Notes/pull/162/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (trgardos) [inline ds701_book/jupyter_notebooks/19-Regression-III-More-Linear.ipynb:7]: <p>I find the description of Patsy confusing. Perhaps we should quote directly from patsy package "<code><a href="https://patsy.readthedocs.io/en/latest/API-reference.html#module-patsy" target="_blank">patsy</a></code>&nbsp;is a Python package for describing statistical models (especially linear models, or models that have a linear component) and building design matrices. It is closely inspired by and compatible with the&nbsp;<a href="http://cran.r-project.org/doc/manuals/R-intro.html#Formulae-for-statistical-models" target="_blank">formula</a>&nbsp;mini-language used in&nbsp;<a href="http://www.r-project.org/" target="_blank">R</a>&nbsp;and&nbsp;<a href="https://secure.wikimedia.org/wikipedia/en/wiki/S_programming_language" target="_blank">S</a>."</p><p>From: <a href="https://patsy.readthedocs.io/en/latest/overview.html" target="_blank">https://patsy.readthedocs.io/en/latest/overview.html</a></p><p>See also <a href="https://www.statsmodels.org/stable/example_formulas.html" target="_blank">https://www.statsmodels.org/stable/example_formulas.html</a></p><p> </p><p>Also, should we introduce patsy where we first use it above? And maybe explain a little more about the syntax? This section feels incomplete -- i know it is from the original notes.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tools4ds/DS701-Course-Notes/pull/162/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='30'/>

**Review by trgardos — CHANGES_REQUESTED** [review]
Overall, looks good, but see my comments.

File: ds701_book/jupyter_notebooks/19-Regression-III-More-Linear.ipynb (hunk 1 above)
Comment (trgardos) [reply ds701_book/jupyter_notebooks/19-Regression-III-More-Linear.ipynb:7]: <p>I suggest printing y.head() too.</p>

Comment (sladenheim) [reply ds701_book/jupyter_notebooks/19-Regression-III-More-Linear.ipynb:7]: <p>Done. y is a series so doesn't have a column name.</p>

Comment (sladenheim) [reply ds701_book/jupyter_notebooks/19-Regression-III-More-Linear.ipynb:7]: <p>Added.</p>

Comment (sladenheim) [reply ds701_book/jupyter_notebooks/19-Regression-III-More-Linear.ipynb:7]: <p>Removed linear function.</p><p> </p><p>Added a comment about scaling as an aside.</p>

Comment (sladenheim) [reply ds701_book/jupyter_notebooks/19-Regression-III-More-Linear.ipynb:7]: <p>Reworked this section to explain what is happening with the patsy syntax.</p>

Comment (sladenheim) [reply ds701_book/jupyter_notebooks/19-Regression-III-More-Linear.ipynb:7]: <p>Reworked the section to include these remarks (see above as well).</p>

**Review by trgardos — APPROVED** [review]

**Pull request merged**